dev/io.openliberty.org.jboss.resteasy.common/src/org/jboss/resteasy/core/ServerResponseWriter.java

```diff
@@ -0,0 +1,801 @@
+package org.jboss.resteasy.core;
+
+import org.jboss.resteasy.core.interception.jaxrs.AbstractWriterInterceptorContext;
+import org.jboss.resteasy.core.interception.jaxrs.ContainerResponseContextImpl;
+import org.jboss.resteasy.core.interception.jaxrs.ResponseContainerRequestContext;
+import org.jboss.resteasy.core.interception.jaxrs.ServerWriterInterceptorContext;
+import org.jboss.resteasy.core.registry.SegmentNode;
+import org.jboss.resteasy.resteasy_jaxrs.i18n.LogMessages;
+import org.jboss.resteasy.specimpl.BuiltResponse;
+import org.jboss.resteasy.spi.AsyncOutputStream;
+import org.jboss.resteasy.spi.HttpRequest;
+import org.jboss.resteasy.spi.HttpResponse;
+import org.jboss.resteasy.spi.ResteasyDeployment;
+import org.jboss.resteasy.spi.ResteasyProviderFactory;
+import org.jboss.resteasy.tracing.RESTEasyTracingLogger;
+import org.jboss.resteasy.util.CommitHeaderAsyncOutputStream;
+import org.jboss.resteasy.util.CommitHeaderOutputStream;
+import org.jboss.resteasy.util.CommitHeaderOutputStream.CommitCallback;
+import org.jboss.resteasy.util.HttpHeaderNames;
+import org.jboss.resteasy.util.MediaTypeHelper;
+
+import javax.ws.rs.NotAcceptableException;
+import javax.ws.rs.Produces;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.NewCookie;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.MessageBodyWriter;
+import javax.ws.rs.ext.WriterInterceptor;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Consumer;
+
+/**
+ * @author <a href="mailto:bill@burkecentral.com">Bill Burke</a>
+ * @version $Revision: 1 $
+ */
+public class ServerResponseWriter
+{
+ // Liberty change begin
+    // caches for static MediaTypes from the API - saves the cost of building a new instance for every response 
```

Comment:
Please include a link to the RESTEasy issue you create for this improvement in this comment.  Thanks

Reply:
Chatted with Gary.  This can be disregarded